Changes to streamline-io

anitmangal opened the pull request:
Closes #5 
- Changed module imports:
	Since pyplot was no longer required in chat_functions.py, removed.
	Since pyplot was required in chat_analyzer.py, added import command.

- Changed function 'activity' in chat_functions.py:
	Shifted all print and pyplot statements along with logic and related attributes to chat_analyzer.py

- Changed function 'interaction_curve' in chat_functions.py:
	Shifted all print and pyplot statements along with logic and related attributes to chat_analyzer.py

anitmangal committed:
Changes to streamline-io

- Changed module imports:
	Since pyplot was no longer required in chat_functions.py, removed.
	Since pyplot was required in chat_analyzer.py, added import command.

- Changed function 'activity' in chat_functions.py:
	Shifted all print and pyplot statements along with logic and related attributes to chat_analyzer.py

- Changed function 'interaction_curve' in chat_functions.py:
	Shifted all print and pyplot statements along with logic and related attributes to chat_analyzer.py

JRavi2 commented:
@anitmangal Thanks for the PR! This is a nice start!
Right now the results are getting printed out without much formatting.
Could you change that to tabulated data? You can refer to the already implemented [percentage part](https://github.com/JRavi2/Chat-Analyzer/blob/streamline-io/src/chat_analyzer.py#L215-L219) as an example.

anitmangal commented:
Yes I will do that. I think only the 'activity' path requires the tabulated data formatting or is there something else as well?

> @anitmangal Thanks for the PR! This is a nice start! Right now the results are getting printed out without much formatting. Could you change that to tabulated data? You can refer to the already implemented [percentage part](https://github.com/JRavi2/Chat-Analyzer/blob/streamline-io/src/chat_analyzer.py#L215-L219) as an example.

anitmangal committed:
Update chat_analyzer.py

- Added Tabular form for activity path if username not defined

JRavi2 commented on src/chat_analyzer.py:
```diff
@@ -230,11 +231,65 @@ def controller(
+        if username:
+            list_activity, list_graph = check_activity(msgs, username, start_date, end_date, show_graph)
+            print('The user {} mostly stays active around {} Hours'.format(list_activity[0], list_activity[1]))
+            if show_graph and globals.CAN_SHOW_GRAPH:
```
Please put an empty line between each of these three lines. This would help separate logic and make the code look a bit cleaner.

JRavi2 commented on src/chat_analyzer.py:
```diff
@@ -230,11 +231,65 @@ def controller(
+            users_act = [["User", "Hours Active"]]
+            for user in user_count: users_act.append([user, user_count[user]['max']])
+            print(tabulate(users_act, headers='firstrow', tablefmt='fancy_grid', colalign=('center', 'center'), floatfmt='.4f'))
+            # For Graph
```
Same for these. Let's space these out.

JRavi2 commented on src/chat_analyzer.py:
```diff
@@ -230,11 +231,65 @@ def controller(
+            plt.xlabel('Time of day (in Hours)')
+            plt.ylabel('Message Count')
+            plt.title('Activity of each user')
+            plt.show()
```
This logic should be within the if condition, else the graph will be shown irrespective of whether the user passed the -sG flag or not.

JRavi2 commented on src/chat_analyzer.py:
```diff
@@ -230,11 +231,65 @@ def controller(
+            'Your' if username else 'The',
+            'decreased' if slope_sign_pred < 0 else 'increased'
+        ))
+        #Graph
```
Small detail, but, there should be a space between the "#" and the comment itself, and an empty line above it.

JRavi2 reviewed (commented):
This is looking good! Just a few small nitpicks before merging. Please see the inline comments below.

JRavi2 commented:
Also, I'd recommend running `flake8 .` from the src directory. This will list any linting errors.
Most linting errors can be fixed by autopep8 running `autopep8 -i -r .` from the src directory. Others would have to be fixed manually.
Both [flake8](https://pypi.org/project/flake8/) and [autopep8](https://pypi.org/project/autopep8/) can be installed using pip.

anitmangal committed:
Changes to  Streamline-io 3

- Fixed an indentation error which allowed graph to show when not asked in activity path

- Increased comprehensibility of source code

- Fixed linting errors using flake8 and autopep8

anitmangal commented:
Follow up on #16 and #17 addressing #5 
- Fixed an indentation error which allowed graph to show when not asked in activity path

- Increased comprehensibility of source code

- Fixed linting errors using flake8 and autopep8

JRavi2 commented:
This looks good!
One final thing left. Since the output for some of the options has changed, the tests need to be updated. To do that you just need to run the program for the test files and put in the updated output in the `expected_res` in [test.py](https://github.com/anitmangal/Chat-Analyzer/blob/streamline-io/tests/tests.py), like [here](https://github.com/anitmangal/Chat-Analyzer/blob/streamline-io/tests/tests.py#L16) and verify that all the tests pass.

Also, no need to worry about the linting for the tests.py file.

anitmangal committed:
Update tests.py

- Updated tests.py to reflect the changes done in output

anitmangal commented:
Follow up on issue #5 
- Updated tests.py to reflect the changes done for output of chat_analyzer.py

JRavi2 commented on src/globals.py:
```diff
@@ -12,6 +12,3 @@ def init() -> None:
         print('Warning: Tkinter is not installed, graphs will not be shown')
         CAN_SHOW_GRAPH = False
-
-    HOURS_LIST = ['00', '01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11',
```
Why did you remove this?

anitmangal replied:
It came in as a linting error in flake8 as it wasn't being used in the program.

JRavi2 commented on src/chat_analyzer.py:
```diff
@@ -230,11 +231,72 @@ def controller(
+                plt.plot(list_graph[0], list_graph[1])
+                hours = ['00', '01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11',
+                         '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23']
+                plt.xticks(ticks=np.arange(24), labels=hours)
```
The `HOURS_LIST` from globals should be used here like it was [previously being used](https://github.com/JRavi2/Chat-Analyzer/pull/16/files#diff-42db47af89ce6cfd4f4994c4b00c4efdffec946c80f8f26238b40e25339f2200L164).

anitmangal replied:
Every time I would run the program, it would return an error stating something around 

> module 'globals' has no attribute 'HOURS_LIST'

JRavi2 commented on src/chat_analyzer.py:
```diff
@@ -230,11 +231,72 @@ def controller(
+                    plt.plot(hours, counts, label=user)
+                hours = ['00', '01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11',
+                         '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23']
+                plt.xticks(ticks=np.arange(24), labels=hours)
```
Same thing here.

anitmangal replied:
Okay, let me try to revert these effects and then I'll update the request.

JRavi2 reviewed (commented):
Just noticed a few things. Please see the inline comments below.

anitmangal committed:
Minor changes to streamline-io

- Added hours_list as a global variable from globals and implemened it in the program

anitmangal commented:
Follow Up
- Added hours_list as a global variable from globals and implemened it in the program.
[Turns out that globals.py used hours_list as global variable but the actual variable was named HOURS_LIST, fixed.]

JRavi2 commented:
Can you capitalize the `hours_list` variable (to `HOURS_LIST`)? It's considered a good convention to keep the global constants capitalized.

anitmangal committed:
Minor minor changes to streamline-io

- Replaced hours_list with HOURS_LIST and removed temporary comments

anitmangal commented:
Follow Up
- Replaced hours_list with HOURS_LIST and removed temporary comments

JRavi2 commented:
Looking all good and merging at last!
Thank you for your patience!
And, congrats on getting your first PR merged :tada:
